### PR TITLE
Remove pins for some Python dependencies

### LINKED ISSUE #25340: [Serve] Unpin (or at least weaken) the uvicorn version dependency?
### Description

Is there any reason that uvicorn is pinned to version 0.16? The most recent version on PyPI is 0.17.6 now. It would be nice to have versioning be more flexible, so either requiring 0.16 or later (but perhaps limiting it to not including a full-blown 1.0 release or 0.18 if you want to be more conservative).

I can try to see if the tests pass with uvicorn 0.17.6 but I'm currently trying to figure out how to build the project locally so that I can run tests :)

Thought I'd just ask here about the version pinning in case you all have an answer already.

Thanks!

### Use case

I'd like to use ray[serve] with more recent versions of uvicorn.

### PR DESCRIPTION
**Pull request by tupui**
## Why are these changes needed?

Remove a few pins on Python to unlock new features/fixes from these libraries.

## Related issue number

Closes #25340

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/requirements.txt [hunk 1]
```diff
@@ -76,20 +76,20 @@ pexpect
 Pillow; platform_system != "Windows"
 pygments
 pyspark==3.1.2
-pytest==5.4.3
-pytest-asyncio==0.16.0
+pytest<7
```
Comment (tupui) [inline python/requirements.txt:79]:
```suggestion
pytest<6
```

File: python/requirements/ml/requirements_tune.txt [hunk 2]
```diff
@@ -20,8 +20,7 @@ hyperopt==0.2.5
 jupyter==1.0.0
 kubernetes==17.17.0
 lightgbm==3.2.1
-matplotlib==3.3.4; python_version < '3.7'
-matplotlib==3.4.3; python_version >= '3.7'
+matplotlib!=3.4.3
```
Comment (tupui) [inline python/requirements/ml/requirements_tune.txt:23]: To mimic setup.py.

**Comment by tupui** [comment]
@pcmoritz it looks like CI is happy with these changes.

**Comment by tupui** [comment]
Bump

**Comment by tupui** [comment]
Up

**Review by stephanie-wang — APPROVED** [review]

**Comment by pcmoritz** [comment]
I'm merging in master to re-run the tests since they haven't run for a while :)

**Review by pcmoritz — APPROVED** [review]

**Comment by pcmoritz** [comment]
@simon-mo Can you have a look at the uvicorn update? Do you know why it was pinned?

**Comment by simon-mo** [comment]
Looks fine as long as test passes. In the past we need this due to incompatibility at release time: https://github.com/ray-project/ray/pull/21612

**Review by richardliaw — APPROVED** [review]

**Pull request merged**